### DISCUSSION TITLE
Community effort: Fine tune object-recognition

### DISCUSSION THREAD

**Discussion by arch-user-france1** [nextcloud/recognize#94] [General]
Make an option to train the AI (So I guess this is possible? I already heard something from Google you cab train)

It has many false-positives (except for cats) so I would like to train the model (maybe you also should ad an option for the Administrator to allow the users to press 'this is accurate' or 'this is a false positive'

**Comment by marcelklehr** [comment]
Hey! Thanks for your feedback! There are indeed a few false positives, still.
Fine-tuning the models in-vivo is beyond the scope of this app at the moment, I'm afraid.
However, there's a file with thresholds for each internal label that you can fine-tune (and hopefully contrinbute back to this repo :) ) as follows

1. Run classifier manually: `node recognize/src/classifier_imagenet.js path/to/your/false/positive/image/here.jpg`
2. Inspect the output and [the rules](https://github.com/marcelklehr/recognize/blob/master/src/rules.yml) to see which label is the false positive
3. Adjust the relevant thresholds in [the rules](https://github.com/marcelklehr/recognize/blob/master/src/rules.yml)
4. Run classifier agin, to check if it worked: `node recognize/src/classifier_imagenet.js path/to/your/false/positive/image/here.jpg`

**Comment by arch-user-france1** [comment]
So this is the 'solution'?
Thank you for that....
So does this train the model or not?

PS: Tomorrow I am back home so I'll look what you really mean

**Comment by marcelklehr** [comment]
Please be civil and note that I'm working on this in my free time.

You are free to use a different app or write one yourself. You are even allowed to contribute code or other resources, or take this app and make a new one out of it.

To your question:

> So does this train the model or not?

This does not strictly speaking train the model, but it let's the app know at which point specific labels spit out by the model are useful.

**Comment by arch-user-france1** [comment]
Still a bit unhandy ...

Is it possible to display it right on the picture for Nextcloud? I don't know how Nextcloud works

**Comment by marcelklehr** [comment]
This is how it is at the moment.

If you like I can also setup a file drop where you can upload false-positive images and we will examine them to adjust the thresholds.

**Comment by arch-user-france1** [comment]
Yeah better but there's one problem: 
1. You have to do it
2. There are privacy relatwd issues

Rather set up something where the user can choose on Nextcloud what the picture really is

So for they go to a picture that is correct and say 'correct'
If the Picture isn't correct they can choose wich item it is.

The only problem is the Maschine learning. It needs an AI that is processing all that information - that's just a lot of work

**Comment by arch-user-france1** [comment]
If anybody is doing that - I apprechiate it

I didn't work with AIs at the moment so I would have to learn a lot (This is now out-to-date, I just edited it to ... correct my Enlish)
